### ISSUE TITLE
Add raw-track-access mode support for ECKD

### ISSUE THREAD

**Issue by viccross** [openmainframeproject/feilong#714]
Currently, ECKD disk support in Feilong is handled the same as FBA disk, where the image is of the filesystem level (for FBA disks this is the only sensible approach).  The Linux DASD driver for ECKD supports a different mode, called raw-track-access mode.  In this mode, the raw count-key-data records of the disk can be accessed.

If this support were introduced, the following benefits could be achieved:

- No need to `dasdfmt` a disk before writing an image to the disk, which would save a lot of time when creating a guest;
- Images of other OSes (z/OS, z/VSE, z/VM) could be supported by Feilong.

Feilong would need to track the type of an ECKD image, so that an image created using raw-track-access mode would be correctly restored.  I am not aware that this could be done automatically, so it would be up to the user registering the image with Feilong to correctly "mark" it with the appropriate type.

I'm happy to start looking at how this could be implemented, since I have experience working with raw-track-access mode as part of the z/VM ESI project. 

As discussed with @mfriesenegger at VM Workshop 2023.

**Comment by viccross** [comment]
It looks like some raw-track-access mode support is already present in `zthin`: `zthinshellutils` already supports raw_track_access mode as an option in the `connectDisk` function, and both `creatediskimage` and `unpackdiskimage` appear to use it.  Now moving on to understanding how these capabilities are utilised by the API server.

**Comment by bjhuangr** [comment]
Yes, raw-track-access mode was used in ckdencode and ckddecode when create/deploy the eckd disk image . Not sure in feilong API - guest_deploy, whether we skip the dasdfmt before eckd image deploy.

**Comment by bjhuangr** [comment]
> No need to `dasdfmt` a disk before writing an image to the disk
Normally, the target disk is larger than the raw disk image size. Still need dasdfmt for the extended region of the disk.
